src/Stratis.Bitcoin/Builder/Feature/FeatureRegistration.cs

```diff
@@ -49,6 +51,13 @@ public interface IFeatureRegistration
         /// <returns>This interface to allow fluent code.</returns>
         IFeatureRegistration DependOn<TImplementation>() where TImplementation : class, IFullNodeFeature;
 
+        /// <summary>
+        /// Adds a feature type list where one of the feature to be used by the feature registration. 
+        /// </summary>
+        /// <param name="featureList">List of feature types</param>
+        /// <returns>This interface to allow fluent code.</returns>
+        IFeatureRegistration DependOn(IEnumerable<Type> featureList);
```

Comment:
This is not exactly what I had in mind, if both `PowConsensusFeature` and `PosConsensusFeature` use inheritance from a common `abstract ConsensusFeature` then we dont need this funky magic, and just make sure there is a dependecy on consensus.